django/contrib/admin/options.py

```diff
@@ -561,14 +576,33 @@ def wrapper(*args, **kwargs):
         urlpatterns = [
             url(r'^$', wrap(self.changelist_view), name='%s_%s_changelist' % info),
             url(r'^add/$', wrap(self.add_view), name='%s_%s_add' % info),
+            url(r'^autocomplete/(?P<field_name>.+)/$',
+                wrap(AutocompleteJsonView.as_view(
+                    admin_site=self.admin_site,
+                    source_model=self.model,
+                    source_model_admin=self,
+                )), name='%s_%s_autocomplete' % info),
             url(r'^(.+)/history/$', wrap(self.history_view), name='%s_%s_history' % info),
             url(r'^(.+)/delete/$', wrap(self.delete_view), name='%s_%s_delete' % info),
             url(r'^(.+)/change/$', wrap(self.change_view), name='%s_%s_change' % info),
+        ]
+        for inline in self.get_inline_instances(request=None):
```

Comment:
Damn it, there is clear way to get this working. Ie there is no reliable way to determine the source_model_admin for a given URL :(

I think we might have to relax permissions a bit and provide the autocomplete to any user for every model which has search fields defined and where the user has any permissions on it.